### DISCUSSION TITLE
MPCTracker hidden walls/barriers

### DISCUSSION THREAD

**Discussion by RocketFan** [ctu-mrs/mrs_uav_system#132] [Q&A] [answered]
Hi,

I started working with your system a week ago, but I came to a weird issue.
When MPCTracker is activated, there exist some invisible walls/barriers, which prevent drones to go through them. Is there any parameter responsible for that limits?

Thanks a lot for your help!
Best, Kuba

**Comment by DanHert** [comment, marked as answer]
Hello, you have discovered the safety area. In a real-world experiment, we define a safety area that the UAV cannot leave. This is also present in the simulation. The safety area is defined in a .yaml world file. By default, we store wolds in mrs_uav_general/config/worlds,
And the default one used for simulation is this one - https://github.com/ctu-mrs/mrs_uav_general/blob/master/config/worlds/world_simulation.yaml

You can disable the safety area completely (use_safety_area: false), or just change the dimensions.

Cheers, Dan H.

**Reply by RocketFan** [reply]
Hello, thank you for your reply, that was exactly what I was looking for.
